### PR TITLE
fix: use NodeJS 14 for all packaged custom resources

### LINKED ISSUE #13484: (@aws-cdk/aws-s3): use lambda.Runtime.NODEJS_12_X by default in Provider
This issue has already been discussed in #13225,   
but there is the line in also `@aws-cdk/aws-s3` that needs to be changed.

https://github.com/aws/aws-cdk/blob/e09250bc92c62cb8ee0a8706ce90d0e82faf2d84/packages/%40aws-cdk/aws-s3/lib/notifications-resource/notifications-resource-handler.ts#L85

### Use Case
Reasons for recommending NODEJS_12_X:

* Node.js 12.x went LTS in October 2019, and has been actively supported since October 2020.
* Node.js 10.x is going end-of-life on April 30th, 2021
https://endoflife.date/nodejs

### Proposed Solution
Use lambda.Runtime.NODEJS_12_X by default in Provider

### Other
* Relates to  #13225

This is a :rocket: Feature Request

### LINKED ISSUE #13534: cdk-core: custom resource framework lambda functions created with deprecated version nodejs10.x
Custom Resource Framework AWS Lambda functions are created with a [deprecated version nodejs10.x](https://github.com/aws/aws-cdk/blob/4c63f09f1e9644877eaffbe78eede3854bec08ab/packages/%40aws-cdk/custom-resources/lib/provider-framework/provider.ts#L191).
CDK users are bugged with AWS Trusted Advisor messages like:
AWS Lambda Functions Using Deprecated Runtimes [...], Deprecation date: 04/30/2021

### Reproduction Steps

Use new CustomResource(this, 'esRequestsResource', {...}) from '@aws-cdk/core'.

### What did you expect to happen?

Use LTS version 14 as referred in https://github.com/nodejs/Release#release-schedule

### Environment

  - **CDK CLI Version  :** 1.92.0 (build de536cc)
  - **Framework Version:**
  - **Node.js Version:** v12.13.0
  - **OS               :** AWS Lambda
  - **Language (Version):** all

### Other

--- 

This is :bug: Bug Report

### PR DESCRIPTION
**Pull request by rix0rrr**
NodeJS 10 is reaching end of life soon, it's time to move to NodeJS 14.

Fixes https://github.com/aws/aws-cdk/issues/13225 
Fixes #13534
Fixes #13484

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Comment by iliapolo** [comment]
@rix0rrr I've added a linked issue to the body I believe this PR resolves. 

Also - maybe cleanup this as well while you're at it? :) https://github.com/aws/aws-cdk/issues/13484

**Comment by iliapolo** [comment]
Also - just for the sake of the link. This overlaps with https://github.com/aws/aws-cdk/pull/12808

**Comment by njlynch** [comment]
Tagging #13488 as well.

Might it be worth upgrading all the way to Node 14 while we're at it? 12 is going to be EOL in about a year.

**Comment by jogold** [comment]
For `tasks.EvaluateExpression` and `ses.DropSpamReceiptRule` it has already been done in #13267 and #13266.

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts [hunk 1]
```diff
@@ -110,6 +110,22 @@ function createEvalFn(runtime: lambda.Runtime, scope: Construct) {
         lambdaPurpose,
         code: lambda.Code.fromAsset(path.join(__dirname, 'eval-nodejs-handler')),
       });
+    case lambda.Runtime.NODEJS_12_X:
```
Comment (njlynch) [inline packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts:113]: ❗  The case statement above means these sections will never run.

**Review by njlynch — DISMISSED** [review]

File: packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts (hunk 1 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-stepfunctions-tasks/lib/evaluate-expression.ts:113]: `goto fail`. Thanks!

**Pull request merged**